discovery-provider/src/tasks/index.py

```diff
@@ -1062,6 +1060,12 @@ def update_task(self):
     web3 = update_task.web3
     redis = update_task.redis
 
+    latest_indexed_block = redis.get(most_recent_indexed_block_redis_key)
+    final_poa_block = helpers.get_final_poa_block(update_task.shared_config)
+    if latest_indexed_block >= final_poa_block:
```

Comment:
identity config has a default config to return 0... in which case this would evaluate to true every time.
Does that seem risky?

Seems like we should either leave default to `null` or add a check to be like `final_poa_block > 0 and ...`